packages/react/src/components/AccountSettings/types.ts

```diff
@@ -12,17 +14,27 @@ import {
  * Note that `PrimitiveProps` is used to get native html types, like `onSubmit`.
  */
 type CommonPasswordFieldProps = PrimitiveProps<PasswordFieldProps, 'input'>;
+type CommonTextFieldProps = PrimitiveProps<TextFieldProps, 'input'>;
+type CommonImageProps = PrimitiveProps<ImageProps, 'img'>;
 type CommonButtonProps = PrimitiveProps<ButtonProps, 'button'>;
 type CommonAlertProps = PrimitiveProps<AlertProps, 'div'>;
 
-/**
+/*
  * These are overridable component types (e.g. submit button).
  */
 export type AccountSettingsPasswordField<Props = {}> = React.ComponentType<
   // `Props` generic allows additional props passed on override components
   Props & CommonPasswordFieldProps & { validationErrors?: string[] }
 >;
 
+export type AccountSettingsTextField<Props = {}> = React.ComponentType<
+  Props & CommonTextFieldProps
+>;
+
```

Comment:
I like this pattern, I've been using it with File Uploader

Reply:
It's pretty clever!